master/internal/trial.go

```diff
@@ -63,9 +63,15 @@ const (
 
 // Trial-specific actor messages.
 type (
-	killTrial    struct{ timedOut bool }
+	killTrial    struct{}
 	restoreTrial struct{}
 
+	// When we issue a TERMINATE workload, we send a delayed terminateTimeout message with a record
+	// of the number of resets that the trial had at the time we issued the TERMINATE. If we
+	// receive the terminateTimeout message and t.resets has not changed, we forcibly kill the
+	// running containers.
+	terminateTimeout struct{ resets int }
```

Comment:
non-blocking: I think calling this `runID` would make it clearer on what this field represents. It also may be clearer that it generalizes to future messages that we send. It may make sense to add it to any messages that we send ourselves.

In our case the `runID` is an autoincrementing int, but hypothetically it could be a uuid if we wanted. So then basically we're saying that if we receive a message that came from a previous `runID` we are going to ignore the message. This is a very similar concept as optimistic locking in DynamoDB: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBMapper.OptimisticLocking.html.

Reply:
I like that name, we can go with that.

Reply:
I like that name, we can go with that.